src/control/cmd/daos_agent/fabric.go

```diff
@@ -125,17 +144,24 @@ func (n *NUMAFabric) GetDevice(numaNode int, netDevClass uint32) (*FabricInterfa
 	n.mutex.Lock()
 	defer n.mutex.Unlock()
 
-	if n.getNumNUMANodes() == 0 {
-		n.log.Infof("No fabric interfaces found, using default interface %q", DefaultFabricInterface.Name)
-		return DefaultFabricInterface, nil
+	// If the only device is the loopback, it may not match the server's requested type, but
+	// we can try it anyway, in case the server is listening on the same machine.
```

Comment:
Similar to Tom, I have some concerns about this approach. There is some precedent, in that dmg is configured to try localhost if there aren't any servers in its configuration, and the goal there was to make things "just work" for folks who are kicking the tires.

My primary concern is that we should make it very clear what's going on here so that there aren't mysterious failures. Under what circumstances would the server be configured to use a non-lo interface and yet still communicate successfully with a client on loopback?

Reply:
If the server is on localhost but configured to use an outward-facing like eth0, lo is still able to communicate with it. There's a docker NLT test case that does this. I'm actually not sure why the agent in that test can't see eth0, but it can't. I'm guessing it's something in docker.